sympy/plotting/plot.py

```diff
@@ -908,176 +943,195 @@ def __init__(self, parent):
 class MatplotlibBackend(BaseBackend):
     def __init__(self, parent):
         super(MatplotlibBackend, self).__init__(parent)
-        are_3D = [s.is_3D for s in self.parent._series]
         self.matplotlib = import_module('matplotlib',
             __import__kwargs={'fromlist': ['pyplot', 'cm', 'collections']},
             min_module_version='1.1.0', catch=(RuntimeError,))
         self.plt = self.matplotlib.pyplot
         self.cm = self.matplotlib.cm
         self.LineCollection = self.matplotlib.collections.LineCollection
-        if any(are_3D) and not all(are_3D):
-            raise ValueError('The matplotlib backend can not mix 2D and 3D.')
-        elif not any(are_3D):
-            self.fig = self.plt.figure()
-            self.ax = self.fig.add_subplot(111)
-            self.ax.spines['left'].set_position('zero')
-            self.ax.spines['right'].set_color('none')
-            self.ax.spines['bottom'].set_position('zero')
-            self.ax.spines['top'].set_color('none')
-            self.ax.spines['left'].set_smart_bounds(True)
-            self.ax.spines['bottom'].set_smart_bounds(False)
-            self.ax.xaxis.set_ticks_position('bottom')
-            self.ax.yaxis.set_ticks_position('left')
-        elif all(are_3D):
-            ## mpl_toolkits.mplot3d is necessary for
-            ##      projection='3d'
-            mpl_toolkits = import_module('mpl_toolkits',
+
+        if isinstance(self.parent, Plot):
+            nrows, ncolumns = 1, 1
+            series_list = [self.parent._series]
+        elif isinstance(self.parent, PlotGrid):
+            nrows, ncolumns = self.parent.nrows, self.parent.ncolumns
+            series_list = self.parent._series
+
+        self.ax = []
+        self.fig = self.plt.figure()
+
+        for i, series in enumerate(series_list):
+            are_3D = [s.is_3D for s in series]
+
+            if any(are_3D) and not all(are_3D):
+                raise ValueError('The matplotlib backend can not mix 2D and 3D.')
+            elif all(are_3D):
+                # mpl_toolkits.mplot3d is necessary for
+                # projection='3d'
+                mpl_toolkits = import_module('mpl_toolkits',
```

Comment:
Here, it doesn't look like you are using mpl_toolkits nor checking if it worked (or maybe import_module throws an exception in that case?).